arbor/threading/threading.cpp

```diff
@@ -87,23 +92,59 @@ void task_system::run(priority_task ptsk) {
     ptsk.run();
 }
 
-void task_system::run_tasks_loop(int i) {
-    auto guard = util::on_scope_exit([] { current_task_queue_ = -1; });
-    current_task_queue_ = i;
+#define HWLOC(exp, msg) if (-1 == exp) throw arbor_internal_error(std::string{"HWLOC Thread failed at: "} + msg);
+
+// If we have found hwloc, pin our thread to a single physical core else noop
+void bind_my_thread(int index, int n_threads) {
+#ifdef ARB_HAVE_HWLOC
+    // Create the topology and ensure we don't leak it
```

Comment:
I am confused - why not using `affinity.hpp`?

Reply:
I am confused as well. Maybe a bad interaction between git and my editor...